src/stan/services/util/initialize.hpp

```diff
@@ -13,202 +13,211 @@
-       * log probability of the model with the unconstrained
-       * parameters once to see if it's valid.
-       *
-       * When at least some of the initialization is random, it will
-       * randomly initialize until it finds a set of unconstrained
-       * parameters that are valid or it hits <code>MAX_INIT_TRIES =
-       * 100</code> (hard-coded).
-       *
-       * Valid initialization is defined as a finite, non-NaN value
-       * for the evaluation of the log probability and all its
-       * gradients.
-       *
-       * @param[in] model the model
-       * @param[in] init a var_context with initial values
-       * @param[in,out] rng random number generator
-       * @param[in] init_radius the radius for generating random values.
-       *   A value of 0 indicates that the unconstrained parameters (not
-       *   provided by init) should be initialized with 0.
-       * @param[in] print_timing indicates whether a timing message should
-       *   be printed to the logger
-       * @param[in,out] logger logger for messages
-       * @param[in,out] init_writer init writer (on the unconstrained scale)
-       * @throws exception passed through from the model if the model has a
-       *   fatal error (not a std::domain_error)
-       * @throws std::domain_error if the model can not be initialized and
-       *   the model does not have a fatal error (only allows for
-       *   std::domain_error)
-       * @return valid unconstrained parameters for the model
-       */
-      template <class Model, class RNG>
-      std::vector<double> initialize(Model& model,
-                                     stan::io::var_context& init,
-                                     RNG& rng,
-                                     double init_radius,
-                                     bool print_timing,
-                                     stan::callbacks::logger& logger,
-                                     stan::callbacks::writer&
-                                     init_writer) {
-        std::vector<double> unconstrained;
-        std::vector<int> disc_vector;
+/**
+ * Returns a valid initial value of the parameters of the model
+ * on the unconstrained scale.
+ *
+ * For identical inputs (model, init, rng, init_radius), this
+ * function will produce the same initialization.
+ *
+ * Initialization first tries to use the provided
+ * <code>stan::io::var_context</code>, then it will generate
+ * random uniform values from -init_radius to +init_radius for missing
+ * parameters.
+ *
+ * When the <code>var_context</code> provides all variables or
+ * the init_radius is 0, this function will only evaluate the
+ * log probability of the model with the unconstrained
+ * parameters once to see if it's valid.
+ *
+ * When at least some of the initialization is random, it will
+ * randomly initialize until it finds a set of unconstrained
+ * parameters that are valid or it hits <code>MAX_INIT_TRIES =
```

Comment:
What is `MAX_INIT_TRIES`?  Is it something the user can set?  If not, then I don't think it should be given a name like this that makes it look like a property that a user could set.